### PR TITLE
Enable Editing of IM Style

### PR DESCRIPTION
**Pull request by gkahiu**
This PR incorporates functionality for defining the filll style of an IM data model as a `QgsFillSymbol`.

image

@vermeulendivan below are a few implementation notes:

- To access the fill symbol object in the IM data model, you can use `implementation_model.fill_symbol()`
- Alternatively, if you are only interested in the colour then you can use `implementation_model.fill_symbol().color()`, which returns a `QColor` object.
- I have extracted the IM colours for the scenario from `0_default_scenario_style.qml` and defined them in `defaults.py`; see the screenshot below. However, please feel free to modify these accordingly.

image

This PR addresses issue #265.

### CODE REVIEW

File: src/cplus_plugin/gui/implementation_model_editor_dialog.py [hunk 1]
```diff
@@ -186,6 +205,7 @@ def _create_implementation_model(self):
             self._implementation_model.description = self.txt_description.toPlainText()
 
         self._layer = self._get_selected_map_layer()
+        self._implementation_model.fill_style = self.fill_symbol_layer().properties()
```
Comment (Samweli) [inline src/cplus_plugin/gui/implementation_model_editor_dialog.py:208]: Even though this is unlikely but lets add a check if the `fill_symbol_layer` functions returns a non-null layer before accessing its `properties`.

**Review by Samweli — CHANGES_REQUESTED** [review]
@gkahiu apart from the one comment I made, everything else looks good to me!

File: src/cplus_plugin/gui/implementation_model_editor_dialog.py (hunk 1 above)
Comment (gkahiu) [reply src/cplus_plugin/gui/implementation_model_editor_dialog.py:208]: By default, the button does not allow nulls

image

But am also checking if there is at least one fill symbol layer in the symbol before saving in the implementation model dialog.

Comment (Samweli) [reply src/cplus_plugin/gui/implementation_model_editor_dialog.py:208]: @gkahiu I meant to have similar check as [this](https://github.com/kartoza/cplus-plugin/pull/268/files#diff-38d09860f74cb1d4956e2ab76f02c0c9d42348ce2380a82ac227ab97c4b39062R184) before accessing the `properties` function. If you look at line 247 you could see that there a potential for the function `fill_symbol_layer` to return `None`.

Comment (gkahiu) [reply src/cplus_plugin/gui/implementation_model_editor_dialog.py]: Hi @Samweli , see the updated PR. I have included the check.

**Review by Samweli — APPROVED** [review]

**Pull request merged**